flamedisx/xenon/x1t_sr1.py

```diff
@@ -251,11 +279,43 @@ def random_truth(self, n_events, fix_truth=None, **params):
         d = super().random_truth(n_events, fix_truth=fix_truth, **params)
 
         # Add extra needed columns
-        # TODO: Add FDC maps instead of posrec resolution
-        d['x_observed'] = np.random.normal(d['x'].values,
-                                           scale=2)  # 2cm resolution)
-        d['y_observed'] = np.random.normal(d['y'].values,
-                                           scale=2)  # 2cm resolution)
+        # x, y, z are the true positions of the event
+        #
+        # x_observed, y_observed are the reconstructed positions with 
+        # field distortion and with posrec smearing but without 
+        # field distortion correction
+        #
+        # x_fdc, y_fdc, z_fdc are the fdc-corrected positions
+        
+        # going from true position to position distorted by field
+        d['r_observed'] = self.drift_field_distortion_map(
+            np.transpose([d['r'].values,
+                          d['z'].values])) 
+        d['x_observed'] = d['r_observed'] * np.cos(d['theta'])
+        d['y_observed'] = d['r_observed'] * np.sin(d['theta'])
+
+        # leave z intact, might want to correct this with drift velocity later
+        d['z_observed'] = d['z']
+
+        # Adding some smear according to posrec resolution
+        d['x_observed'] = np.random.normal(d['x_observed'].values, scale=0.4) # 4 mm resolution)
+        d['y_observed'] = np.random.normal(d['y_observed'].values, scale=0.4) # 4 mm resolution)
+        
+        # applying fdc
+        delta_r = self.fdc_map(
```

Comment:
... and here. My suggestion would be to keep things as they were for 1T and add a flag for nT where positions are corrected by the FDC maps.